lib/spack/spack/fetch_strategy.py

```diff
@@ -792,11 +802,460 @@ def __str__(self):
         return "[go] %s" % self.url
 
 
+class GitRef(object):
+    """A wrapper that knows how to format git refspecs in multiple contexts."""
+    _known_types = ['branch', 'commit', 'tag']  # type: ClassVar[List[str]]
+    _ref_type = None                            # type: str
+    _ref = None                                 # type: str
+
+    @classmethod
+    def known_types(cls):
```

Comment:
A question for my own curiository - what is the benefit to having this getter return cls._known_types vs. just interacting with a variable cls.known_types directly? Is it an effort to try to protect the actual variable from being touched (since it's "private" with an underscore?)

Reply:
In general it is a way to perform "information hiding" by decoupling the accessor from the location of data, but in this case since the accessor isn't doing any validation or whatever, I agree it is unnecessary. I have also removed the unnecessary accessor for `_ref_type`! Thank you!